src/plugins/intel_cpu/src/nodes/executors/acl/acl_fullyconnected.cpp

```diff
@@ -0,0 +1,99 @@
+    fullyConnectedLayerInfo.weights_trained_layout = getAclDataLayoutByMemoryDesc(memory.at(ARG_WEI)->getDescPtr());
+    fullyConnectedLayerInfo.transpose_weights = !attrs.weightsNonTransposed;
+    if (memory.at(ARG_SRC)->getPrecision() == ov::element::f16) {
+        fullyConnectedLayerInfo.fp_mixed_precision = true;
+    }
+
+    // Add postops
+    if (!postOps.empty() && postOps.size() == 1) {
+        if (const auto activation = std::dynamic_pointer_cast<ActivationPostOp>(postOps[0])) {
+            fullyConnectedLayerInfo.activation_info = getActivationLayerInfo(convertToEltwiseAlgorithm(activation->type()),
+                                                                             activation->alpha(),
+                                                                             activation->beta(),
+                                                                             activation->gamma());
+        }
+    }
+}
+
+bool ACLFullyConnectedExecutor::supports(const FCConfig &config) {
+    if (!config.postOps.empty() && config.postOps.size() != 1) {
+        DEBUG_LOG("ACLFullyConnectedExecutor supports only 1 post op");
+        return false;
+    }
+
+    const auto& srcDesc = config.descs.at(ARG_SRC);
+    if (!one_of(srcDesc->getShape().getDims().size(), 2U, 3U, 4U)) {
+        DEBUG_LOG("ACLFullyConnectedExecutor supports only 2, 3 or 4 dimensions for inputs");
+        return false;
+    }
+
+    const auto& weiDesc = config.descs.at(ARG_WEI);
+    if (!one_of(weiDesc->getShape().getDims().size(), 2U, 3U)) {
+        DEBUG_LOG("ACLFullyConnectedExecutor supports only 2 or 3 dimensions for weights");
+        return false;
+    }
+    return true;
+}
+
+void ACLFullyConnectedExecutor::prepareTensorsInfo() {
+    auto wei_shape = aclMemoryInfoMap[ARG_WEI]->tensor_shape();
+    if (wei_shape.num_dimensions() == 3U) {
+        aclMemoryInfoMap[ARG_WEI]->set_tensor_shape({wei_shape[0] * wei_shape[1], wei_shape[2]});
+        wei_shape = aclMemoryInfoMap[ARG_WEI]->tensor_shape();
+    }
+
+    auto src_shape = aclMemoryInfoMap[ARG_SRC]->tensor_shape();
+    if (one_of(src_shape.num_dimensions(), 3U, 4U)) {
+        aclMemoryInfoMap[ARG_SRC]->set_tensor_shape({wei_shape[0], src_shape.total_size() / wei_shape[0]});
+        src_shape = aclMemoryInfoMap[ARG_SRC]->tensor_shape();
+    }
+
+    if (one_of(aclMemoryInfoMap[ARG_DST]->tensor_shape().num_dimensions(), 3U, 4U)) {
+        aclMemoryInfoMap[ARG_DST]->set_tensor_shape({wei_shape[1], src_shape[1]});
+    }
+
+    if (!fullyConnectedLayerInfo.transpose_weights) {
+        arm_compute::TensorShape temp_weights_shape = aclMemoryInfoMap[ARG_WEI]->tensor_shape();
+        std::swap(temp_weights_shape[0], temp_weights_shape[1]);
+        aclMemoryInfoMap[ARG_WEI]->set_tensor_shape(temp_weights_shape);
+    }
+
+    tensorsInfoValidateStatus = arm_compute::NEFullyConnectedLayer::validate(
```

Comment:
Does not oneDNN use weights packing feature for ACL integration?
https://arm-software.github.io/ComputeLibrary/v23.02.1/classarm__compute_1_1_n_e_fully_connected_layer.xhtml#a19aa329510cbef84acc16335c2099908
Just asking.
Because, if not, later we better to try to use it by ourselves.

Reply:
Discussed. oneDNN does use has_opt_impl feature (basically weights packing).
So, the oneDNN logic needs to be replicated for ACLFCExecutor to ensure no performance drop.
We can merge the PR with no weights packing support, as soon as all the tests are passed, but completely disable the ACLFCExecutor for now.